File: Patterns/js/numberStar.js
/*
n = 5

1234554321
1234**4321
123****321
12******21
1********1

*/
function pattern(n) {
  let ans = "";
  let row = 1;
  let starCol = 0;
  while (row <= n) {
    let col = 1;
    while (col <= n - row + 1) {
      ans += col;
      col++;
    }

    col = 1;
    while (col <= starCol) {
      ans += "*";
      col++;
    }

    starCol += 2;

    col = 1;
    let val = n - row + 1;
    while (col <= n - row + 1) {
      ans += val;
      val--;
      col++;
    }

    row++;
    ans += "\n";
  }

  console.log(ans);
}

pattern(5);
pattern(8);

File: Patterns/python/numberStar.py
# n = 5

# 1234554321
# 1234**4321
# 123****321
# 12******21
# 1********1

n = int(input())
row = 1
starCol = 0
while row <= n:
    col = 1
    while col <= n - row + 1:
        print(col, end="")
        col += 1
    
    col = 1
    while col <= starCol:
        print("*", end="")
        col += 1
    
    starCol += 2

    col = 1
    val = n - row + 1
    while col <= n - row + 1:
        print(val, end="")
        val -= 1
        col += 1
    
    row += 1
    print("")